nixos/modules/services/networking/sharkey.nix

```
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.sharkey;
+
+  settingsFormat = pkgs.formats.yaml {};
+  configFile = settingsFormat.generate "sharkey-config.yml" cfg.settings;
+in {
+  options = {
+    services.sharkey = with lib; {
+      enable = mkEnableOption "sharkey";
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = {};
+        description = ''
+          Configuration for Sharkey, see
+          <link xlink:href="https://activitypub.software/TransFem-org/Sharkey/-/blob/develop/.config/example.yml"/>
+          for supported settings.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    documentation.enable = false;
+
+    systemd.services.sharkey = {
+      after = [ "network-online.target" "postgresql.service" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        ${pkgs.envsubst}/bin/envsubst -i "${configFile}" > /run/sharkey/default.yml
+        cd ${pkgs.sharkey}/data
+      '';
+      serviceConfig = {
```

Is this running as root?
```suggestion
      serviceConfig = {
        User = "sharkey";
```